jib-core/src/main/java/com/google/cloud/tools/jib/image/LayerEntry.java

```diff
@@ -16,37 +16,102 @@
 
 package com.google.cloud.tools.jib.image;
 
-import com.google.common.collect.ImmutableList;
+import com.google.common.annotations.VisibleForTesting;
 import java.nio.file.Path;
+import java.util.Objects;
 
 /**
  * Represents an entry in the layer. A layer consists of many entries that can be converted into tar
  * archive entries.
  */
 public class LayerEntry {
 
+  private final Path sourceFile;
+  private final Path extractionPath;
+
   /**
-   * The source files to build from. Source files that are directories will have all subfiles in the
-   * directory added (but not the directory itself).
+   * Instantiates with a source file and the path to place the source file in the container file
+   * system. For example, adding a file {@code HelloWorld.class} to be in container file system at
+   * {@code /app/classes/HelloWorld.class} would be {@code new
+   * LayerEntry(Paths.get("HelloWorld.class"), Paths.get("/app/classes/HelloWorld.class"))}.
```

Comment:
You could add an example here of `com` &rarr; `/app/classes/com` but not its contents.